src/App.jsx

```diff
@@ -7,139 +7,232 @@
-			this.setState({
-				hash: hash || '#'
-			});
-		});
-
-		apiUtils.checkApiAvailability().then(() => {
-			apiUtils.checkWritePermissions();
-			return apiUtils.initialLoad(datastore);
-		}).then(() => {
-			this.setState({ loading: false });
-			datastore.startObserve();
-		}).catch((err) => {
-			this.setState({ error: err.type });
-		});
-	}
-
-	render() {
-		const { error, loading } = this.state;
-
-		if (loading) {
-			return (
-				<div className={ styles.splash }>
-					<span className={ styles.logo } />
-					<Loader className={ styles.loader } />
-					<span className={ styles.loading }>Loading...</span>
-				</div>
-			);
-		}
-
-		let content;
-
-		if (error) {
-			let subContent = null;
-
-			if (error in Errors) {
-				subContent = <p>{ Errors[error] }</p>;
-			}
-
-			content = (
-				<div className={ styles['error-block'] }>
-					{ subContent }
-					<p>For&nbsp;more information please refer to&nbsp;the&nbsp;following <a href="https://www.nginx.com/resources/admin-guide/monitoring/">documentation.</a></p>
-				</div>
-			);
-		} else {
-			const Page = SECTIONS[this.state.hash];
-
-			content = <Page />;
-		}
-
-		return (
-			<div className={ styles.dashboard }>
-
-				{
-					__ENV__ === 'demo' ?
-						<Disclaimer />
-					: null
-				}
-
-				<Header hash={this.state.hash} navigation={!error} statuses={STORE.__STATUSES} />
-
-				<div className={ styles.content }>
-					{
-						!this.state.error ?
-							<UpdatingControl play={play} pause={pause} update={startObserve} />
-						: null
-					}
-
-					{ content }
-				</div>
-
-				<Footer />
-			</div>
-		);
-	}
+  constructor() {
+    super();
+
+    this.state = {
+      hash: history.location.hash || "#",
+      loading: false,
+      error: undefined,
+      license: undefined,
+    };
+    this.getAlertMessage = this.getAlertMessage.bind(this);
+  }
+
+  componentDidMount() {
+    history.listen(({ hash }) => {
+      this.setState({
+        hash: hash || "#",
+      });
+    });
+
+    apiUtils
+      .checkApiAvailability()
+      .then(() => {
+        apiUtils.checkWritePermissions();
+        api.license
+          .get()
+          .then((result) => {
+            this.setState({ license: res });
+          })
+          .catch((err) => {
+            this.setState({ error: err.type });
+          });
+        return apiUtils.initialLoad(datastore);
+      })
+      .then(() => {
+        this.setState({ loading: false });
+        datastore.startObserve();
+      })
+      .catch((err) => {
+        this.setState({ error: err.type });
+      });
+  }
+
+  getAlertMessage() {
+    let licenseMessage;
+    let usageMessage;
+
+    if (this.state.license) {
+      const expireDate = Date.parse(this.state.license?.active_till);
+      const hasExpired = Date.now() > expireDate;
+      const fifteenDays = 1296000000;
+      const timeTillExpiration = expireDate - Date.now();
+      const plural = Math.round(timeTillExpiration / 86400000) > 1;
```

Comment:
If the number of days in milliseconds is less than 1, 0 days should be displayed - this matches the nginx behavior.
If the number of days is 0, `plural` must be true.
```suggestion
      const daysLeft = Math.floor(timeTillExpiration / 86400000);
      const plural = daysLeft !== 1;
```